pkg/querier/queryrange/querysharding.go

```diff
@@ -252,6 +252,7 @@ func (ast *astMapperware) Do(ctx context.Context, r queryrangebase.Request) (que
 		return nil, err
 	}
 
+	// TODO(tpatterson): add warnings to response
```

Comment:
This `TODO` seems obsolete, you seem to have added the warnings to all of the response types below ?